x/subscription/keeper/epoch_start.go

```diff
@@ -0,0 +1,22 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/x/subscription/types"
+)
+
+// EpochStart runs the functions that are supposed to run in epoch start
+func (k Keeper) EpochStart(ctx sdk.Context) {
+	// On epoch start we need to iterate through all the subscriptions and
+	// check if they just expired. Those that expire get deleted.
+
+	date := ctx.BlockTime().UTC()
+	subExpired := k.GetCondSubscription(ctx, func(sub types.Subscription) bool {
```

Comment:
instead of iterating over all the subscriptions i think we should create an ordered list that contains the next subscription to expire, that way no iterating is required.

Reply:
Deferred to a separate issue.